### PR TITLE
Playback landscape improvement

### LINKED ISSUE #4913: Speedmenu layout in landscape mode
### Checklist

- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one feature request. I will open one issue for every feature I want to request.

#### Describe the feature you want

I think the speed control should have another layout  in landscape mode, so most users don't have to scroll.
There is an image how it could look

#### Is your feature request related to a problem? Please describe it

At the moment you have to scroll through the menu and see not all options at the same time. 
I have to admit that my phone screen may be a bit thin and long but i think it is the same issue on many other devices that the menu is too big in vertical and has enough space left on the sites. 

#### Additional context

At the time:
old_speed-menu
My suggestion:
new_speed-menu

#### How will you/everyone benefit from this feature?

Its just a detail, but i think it would improve UX because you see everything after opening it up and don't have to scroll anymore - it would just feel cleaner.

### PR DESCRIPTION
**Pull request by Hugov8**
#### What is it?
- [ ] Bugfix (user facing)
- [x] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

Improvement of the UX when we are in landscape mode for the speedmenu dialog. Users don't have to scroll down anymore to see all the paramaters like in the screenshots below. A new layout file (dialog_playback_parameter_landscape.xml) has been added and the PlaybackParameterDialog.java has been modified to show the right layout. 2 new constants have been added in values.xml file.

#### Before/After Screenshots/Screen Record

- Before: 
layout_before1
layout_before2

- After: 
layout_after

#### Fixes the following issue(s)

- Fixes #4913 

#### APK testing 

[app-debug.zip](https://github.com/TeamNewPipe/NewPipe/files/7467486/app-debug.zip)

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

File: app/src/main/res/values-iw [hunk 1]
```diff
@@ -1 +0,0 @@
-values-he/
```
Comment (litetex) [inline app/src/main/res/values-iw:1]: Why was that deleted?

File: build.gradle [hunk 2]
```diff
@@ -8,7 +8,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:7.0.2'
+        classpath 'com.android.tools.build:gradle:7.0.3'
```
Comment (litetex) [inline build.gradle:11]: Why was that increased?

File: app/src/main/java/org/schabi/newpipe/player/helper/PlaybackParameterDialog.java [hunk 3]
```diff
@@ -2,9 +2,13 @@
 
 import android.app.Dialog;
 import android.content.Context;
+//import android.content.DialogInterface;
```
Comment (litetex) [inline app/src/main/java/org/schabi/newpipe/player/helper/PlaybackParameterDialog.java:5]: Dead code

File: app/src/main/java/org/schabi/newpipe/player/helper/PlaybackParameterDialog.java [hunk 4]
```diff
@@ -153,11 +157,22 @@ public void onSaveInstanceState(final Bundle outState) {
...
+
+        final int orientation = getResources().getConfiguration().orientation;
+        View view = null;
+        if (orientation == Configuration.ORIENTATION_LANDSCAPE) {
+            view = View.inflate(getContext(),
+                    R.layout.dialog_playback_parameter_landscape, null);
+        } else {
+            view = View.inflate(getContext(), R.layout.dialog_playback_parameter, null);
+
+        }
```
Comment (litetex) [inline app/src/main/java/org/schabi/newpipe/player/helper/PlaybackParameterDialog.java:169]: Can be done inline.

e.g.
```
final View view = View.inflate(
   getContext(),
   getResources().getConfiguration().orientation == Configuration.ORIENTATION_LANDSCAPE
      ? R.layout.dialog_playback_parameter_landscape
      : R.layout.dialog_playback_parameter
   null);
```

**Review by litetex — CHANGES_REQUESTED** [review]
Did a very quick review (see the comments below).

Also some documentation would be good.

I also don't know if it's a good idea to go so far away from the default design of the Dialog.

**Comment by Hugov8** [comment]
Thanks @litetex  for your feedback. I put back the values-iw, I didn't intend to remove this (I don't know why it was on my change list on github desktop) and put back the right version of gradle (I click on upgrade by mistake on the IDE)

Concerning the PlaybackParameterDialog I remove the setupButtonDialog function because it didn't reduce the button area as I expected. I remove the dead code and put in 1 line the instantiation of the view.

However I don't understand what you mean by going far away from the default design of the dialog ? Is the custom title that bothering you or the add of a new layout file or something else ?

**Comment by peat80** [comment]
This pr probably supersedes #6150

File: app/src/main/res/values-iw (hunk 1 above)
Comment (litetex) [reply app/src/main/res/values-iw:1]: Still not as it was before ;)

Comment (Hugov8) [reply app/src/main/res/values-iw]: Ah sorry, I've got some difficulties to put back this one. It should be ok now

**Review by litetex — COMMENTED** [review]
Some things I found while reviewing:
* It's really hard to review if both layouts match or not
* The layout is inconsistent with the original layout e.g. regarding text-alignment
* The color of the title is always black. Doesn't look good when using the black theme.
* There are problems that the checkboxes are completely off-centered when using e.g. German as language

In general this raised the question if a second layout is the best idea here in the first place.
See also https://github.com/TeamNewPipe/NewPipe/issues/4913#issuecomment-961309165

**Comment by Hugov8** [comment]
Here is my new implementation. I keep only one layout file and I rework the marges and padding to reduce the size and gain space. I keep the custom title because it saves a lot of space. This solution is not much different of the #7363 but I put in PlaybackParameter.java a function to change the orientation of the checkboxes area depending the orientation of the phone to save space. It separates well in the middle in English and French but not in German but I don't know how to fix that without complexifing the layout file or the code.

**Comment by litetex** [comment]
Superseded by #7363.

Anyway thank you for the PR 😄

**Comment by Hugov8** [comment]
You're welcome, thanks for taking time to feedback, it was my first contribution and it was very instructive :smiley:
